packages/next/src/lib/metadata/resolvers/resolve-opengraph.ts

```diff
@@ -58,25 +59,41 @@ function resolveAndValidateImage(
   // detect in the build environment if the deployment is a Vercel deployment or not.
   //
   // x-ref: https://vercel.com/docs/projects/environment-variables/system-environment-variables#system-environment-variables
-  const isNonVercelDeployment = !process.env.VERCEL
-  // When not deployed to Vercel, we aren't able to determine a fallback value for `metadataBase`
-  // since it's derived from injected environment variables. For self-hosted setups, we want to warn
-  // about this since the only fallback we'll be able to generate is `localhost`.
-  if (isNonVercelDeployment) {
-    if (
-      typeof inputUrl === 'string' &&
-      !isFullStringUrl(inputUrl) &&
-      !metadataBase
-    ) {
+  const isUsingVercelSystemEnvironmentVariables = Boolean(process.env.VERCEL)
+
+  const isRelativeUrl =
+    typeof inputUrl === 'string' && !isFullStringUrl(inputUrl)
+
+  // We no explicit metadataBase is specified by the user, we'll override it with the fallback metadata
+  // under the following conditions:
+  // - The user didn't provide a metadataBase and the provided URL is relative (ie ./og-image).
```

Comment:
"The user didn't provide a metadataBase" is already mentioned above, kinda redudant here